crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -23,42 +22,88 @@ pub enum ShaderReflectError {
     #[error(transparent)]
     Validation(#[from] naga::WithSpan<naga::valid::ValidationError>),
 }
-/// A shader, as defined by its [`ShaderSource`] and [`ShaderStage`](naga::ShaderStage)
+/// A shader, as defined by its [`ShaderSource`](wgpu::ShaderSource) and [`ShaderStage`](naga::ShaderStage)
 /// This is an "unprocessed" shader. It can contain preprocessor directives.
 #[derive(Debug, Clone, TypeUuid)]
 #[uuid = "d95bc916-6c55-4de3-9622-37e7b6969fda"]
 pub struct Shader {
-    source: Source,
-    import_path: Option<ShaderImport>,
-    imports: Vec<ShaderImport>,
+    pub path: Option<String>,
+    pub source: Source,
+    pub import_path: Option<ShaderImport>,
+    pub imports: Vec<ShaderImport>,
+    // extra imports not specified in the source string
+    pub additional_imports: Vec<naga_oil::compose::ImportDefinition>,
```

Comment:
I'm not sure what this means. Does this mean we could support a kind of "default imports" feature a bit like the rust std prelude that isn't present in the file?

Reply:
it's a set of additional modules that get included in the final built shader. the motivation is for overrides. if a user/plugin wants to include a module in the standard shader with some overrides for core functionality, we would have to either somehow modify the string of the top level shader, or just wholesale replace it (which could only work for one plugin at a time obviously). 

this allows multiple plugins to add functionality in a more modular way, and i hope eventually we'll have some of the current "core" functionality (like cascading shadow maps for example) being hooked in here via modules with overrides.